[v3.52.0 regression] Tests fail when built from source archive

### Description

I was updating the AerynOS package to the latest version when I saw that the test suite now fails. As far as I can tell this is caused by using the Github-generated source archive for building go-task (which is typical for distribution packages as it's much more easily cached) and introduced by 616433df758ca2517334e42651c2ddf5fba57ef5. Likely all of the failing tests should be skipped if running from a non-git source or modified so that they manage their own git directory.

```
+ go test ./...
--- FAIL: TestGitignoreChecksum (0.00s)
    --- FAIL: TestGitignoreChecksum/ignored_file_modified (0.00s)
        task_test.go:725: Result did not match the golden fixture. Diff is below:

            --- Expected
            +++ Actual
            @@ -1,2 +1,2 @@
            -task: Task "build" is up to date
            +task: [build] cp ./source.txt ./generated.txt


--- FAIL: TestGitignoreNegation (0.00s)
    --- FAIL: TestGitignoreNegation/ignored_file_modified (0.00s)
        task_test.go:741: Result did not match the golden fixture. Diff is below:

            --- Expected
            +++ Actual
            @@ -1,2 +1,3 @@
            -task: Task "build" is up to date
            +task: [build] echo "build executed"
            +build executed


--- FAIL: TestGitignoreNested (0.00s)
    --- FAIL: TestGitignoreNested/ignored_file_modified (0.00s)
        task_test.go:758: Result did not match the golden fixture. Diff is below:

            --- Expected
            +++ Actual
            @@ -1,2 +1,3 @@
            -task: Task "build" is up to date
            +task: [build] echo "build executed"
            +build executed


--- FAIL: TestGitignoreIncluded (0.00s)
    --- FAIL: TestGitignoreIncluded/ignored_file_modified (0.00s)
        task_test.go:773: Result did not match the golden fixture. Diff is below:

            --- Expected
            +++ Actual
            @@ -1,2 +1,3 @@
            -task: Task "included:build" is up to date
            +task: [included:build] echo "build executed"
            +build executed


FAIL
FAIL        github.com/go-task/task/v3      0.458s
```

### Version

v3.52.0

### Operating system

Linux (AerynOS)

### Experiments Enabled

_No response_

### Example Taskfile

```YAML

```

--- a/task_test.go
+++ b/task_test.go
@@ -681,6 +681,12 @@ type gitignoreSeq struct {
 func (s gitignoreSeq) run(t *testing.T) {
 	t.Helper()
 	cleanup := func() {
+		// The fixture manages its own .git marker so that gitignore filtering
+		// resolves a repo root regardless of the build source: an in-tree
+		// checkout would otherwise inherit the go-task .git, but a GitHub
+		// source tarball has none, which would silently disable filtering and
+		// break the golden fixtures.
+		_ = os.RemoveAll(filepathext.SmartJoin(s.dir, ".git"))
 		_ = os.RemoveAll(filepathext.SmartJoin(s.dir, ".task"))
 		for name := range s.create {
 			_ = os.Remove(filepathext.SmartJoin(s.dir, name))
@@ -694,6 +700,7 @@ func (s gitignoreSeq) run(t *testing.T) {
 	}
 	cleanup()
 	t.Cleanup(cleanup)
+	require.NoError(t, os.MkdirAll(filepathext.SmartJoin(s.dir, ".git"), 0o755))
 	for name, content := range s.create {
 		writeFile(t, s.dir, name, content)
 	}

